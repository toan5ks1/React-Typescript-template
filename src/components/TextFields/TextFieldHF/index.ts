export { default } from "./TextFieldHF";
export * from "./TextFieldHF";
