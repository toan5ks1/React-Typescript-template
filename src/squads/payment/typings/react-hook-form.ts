import { RegisterOptions, UseFormReturn as UseFormMethodsHF } from "react-hook-form";

//TODO: merge rules with HookFormControllerOptionProps, will create another PR
export interface UseFormRules {
    rules?: Exclude<RegisterOptions, "valueAsNumber" | "valueAsDate" | "setValueAs">;
}

export interface UseFormMethods<T> {
    methods?: UseFormMethodsHF<T>;
}

export interface UseFormType<T = any>
    extends Pick<
            UseFormMethodsHF<T>,
            | "register"
            | "formState"
            | "control"
            | "setValue"
            | "getValues"
            | "watch"
            | "reset"
            | "handleSubmit"
        >,
        UseFormRules {}
