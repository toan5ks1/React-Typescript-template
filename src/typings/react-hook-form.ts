import {
    RegisterOptions,
    UseControllerProps,
    UseFormReturn as UseFormMethodsHF,
} from "react-hook-form";

//TODO: merge rules with HookFormControllerOptionProps, will create another PR
export interface UseFormRules {
    rules?: Exclude<RegisterOptions, "valueAsNumber" | "valueAsDate" | "setValueAs">;
}

export interface UseFormMethods<T> {
    methods?: UseFormMethodsHF<T>;
}

export interface UseFormType<T = any>
    extends Pick<
            UseFormMethodsHF<T>,
            | "register"
            | "formState"
            | "control"
            | "setValue"
            | "getValues"
            | "watch"
            | "reset"
            | "handleSubmit"
        >,
        UseFormRules {}

// TODO: Remove "control" after refactor HFComponent
export interface HookFormCommonProps<T = any>
    extends Pick<UseFormMethodsHF<T>, "register" | "control">,
        Pick<UseFormType<T>, "rules" | "formState" | "setValue" | "getValues" | "watch"> {}

export interface HookFormControllerOptionProps<T = any>
    extends Pick<UseControllerProps<T>, "rules" | "name"> {}
