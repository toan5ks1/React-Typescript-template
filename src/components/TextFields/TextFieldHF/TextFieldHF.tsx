import get from "lodash/get";
import { useFormContext } from "react-hook-form";
import { HookFormControllerOptionProps } from "src/typings/react-hook-form";

import { FormControl } from "@mui/material";

import TextFieldBase, { TextFieldBaseProps } from "../TextFieldBase";

export interface TextFieldHFProps
    extends Omit<TextFieldBaseProps, "onChange">,
        Pick<HookFormControllerOptionProps, "rules"> {
    name: string;
}

const TextFieldHF = ({
    name,
    size = "small",
    rules,
    inputProps,
    FormHelperTextProps,
    ...rest
}: TextFieldHFProps) => {
    const {
        register,
        formState: { errors },
    } = useFormContext();

    const fieldError = get(errors, name);

    return (
        <FormControl variant="outlined" fullWidth size={size}>
            <TextFieldBase
                name={name}
                size={size}
                FormHelperTextProps={{
                    variant: "standard",
                    "aria-label": "TextFieldHF__formHelperText",
                    ...FormHelperTextProps,
                }}
                error={Boolean(fieldError)}
                helperText={Boolean(fieldError) && fieldError["message"]}
                inputProps={{
                    "data-testid": "TextFieldHF__input",
                    ...inputProps,
                    ...register(name, rules),
                }}
                {...rest}
            />
        </FormControl>
    );
};

export default TextFieldHF;
