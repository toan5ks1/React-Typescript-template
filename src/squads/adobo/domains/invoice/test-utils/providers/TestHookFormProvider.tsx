import { PropsWithChildren } from "react";

import { useForm } from "react-hook-form";

import HookForm from "src/components/Forms/HookForm";

export interface TestHookFormProviderProps {
    methodsOverride?: object;
    formProps?: object;
    useFormOptions?: Parameters<typeof useForm>[0];
}
const TestHookFormProvider = ({
    children,
    methodsOverride = {},
    formProps = {},
    useFormOptions,
}: PropsWithChildren<TestHookFormProviderProps>) => {
    const methods = useForm(useFormOptions);
    const { handleSubmit } = methods;
    return (
        <HookForm
            methods={{
                ...methods,
                ...methodsOverride,
            }}
            formProps={{
                onSubmit: handleSubmit((...args: any) => jest.fn(...args)),
                ...formProps,
            }}
        >
            {children}
            <input data-testid="TextFieldHF__submit" type="submit" />
        </HookForm>
    );
};

export default TestHookFormProvider;
