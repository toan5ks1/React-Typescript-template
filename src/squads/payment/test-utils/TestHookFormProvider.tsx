import { PropsWithChildren } from "react";

import { useForm, UseFormProps } from "react-hook-form";

import HookForm from "src/components/Forms/HookForm";

export interface TestHookFormProviderProps<TFieldValues> {
    methodsOverride?: object;
    formProps?: object;
    useFormOptions?: UseFormProps<TFieldValues>;
}
const TestHookFormProvider = <TFieldValues extends object>({
    children,
    methodsOverride = {},
    formProps = {},
    useFormOptions,
}: PropsWithChildren<TestHookFormProviderProps<TFieldValues>>) => {
    const methods = useForm(useFormOptions);
    const { handleSubmit } = methods;
    return (
        <HookForm
            methods={{
                ...methods,
                ...methodsOverride,
            }}
            formProps={{
                onSubmit: handleSubmit((...args: any) => jest.fn(...args)),
                ...formProps,
            }}
        >
            {children}
            <input data-testid="TextFieldHF__submit" type="submit" />
        </HookForm>
    );
};

export default TestHookFormProvider;
