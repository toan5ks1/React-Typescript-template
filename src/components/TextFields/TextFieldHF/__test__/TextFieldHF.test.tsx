import { useForm } from "react-hook-form";

import TextFieldHF from "../TextFieldHF";

import { render, RenderResult, fireEvent } from "@testing-library/react";
import TestHookFormProvider from "src/test-utils/TestHookFormProvider";

const defaultProps = {
    name: "name",
    label: "Name",
    onChange: jest.fn(),
};

const HookFormComponent = (restProps: any) => {
    const methods = useForm();
    const {
        register,
        formState: { errors },
        handleSubmit,
    } = methods;
    return (
        <TestHookFormProvider
            methodsOverride={methods}
            formProps={{
                onSubmit: handleSubmit((...args: any) => jest.fn(...args)),
            }}
        >
            <TextFieldHF
                {...defaultProps}
                {...restProps}
                errors={errors}
                inputProps={register("name", {
                    required: { value: true, message: "Name is required" },
                })}
            />
        </TestHookFormProvider>
    );
};
describe("<TextFieldHF /> required error", () => {
    let wrapper: RenderResult;

    beforeEach(() => {
        wrapper = render(<HookFormComponent />);
    });

    it("should exist error", async () => {
        expect(wrapper.queryByTestId("TextFieldHF__input")).toBeInTheDocument();
        expect(wrapper.queryByTestId("TextFieldHF__submit")).toBeInTheDocument();

        fireEvent.click(wrapper.getByTestId("TextFieldHF__submit"));

        expect(await wrapper.findByText("Name is required")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});

describe("<TextFieldHF /> default value", () => {
    let wrapper: RenderResult;

    beforeEach(() => {
        wrapper = render(<HookFormComponent defaultValue="test data" />);
    });

    it("should exist value", () => {
        expect(wrapper.container.querySelector("input")?.value).toEqual("test data");
        expect(wrapper.queryByText("Name is required")).not.toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
