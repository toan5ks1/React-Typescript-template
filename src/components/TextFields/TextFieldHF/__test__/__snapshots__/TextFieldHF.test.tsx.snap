// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldHF /> default value should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-1 .TextFieldBase-readOnly .MuiInputBase-root {
  background-color: #fafafa;
}

.emotion-2 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(133% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, -9px) scale(0.75);
  -moz-transform: translate(14px, -9px) scale(0.75);
  -ms-transform: translate(14px, -9px) scale(0.75);
  transform: translate(14px, -9px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: auto;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-2.Mui-focused {
  color: #1976d2;
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-2.Mui-error {
  color: #d32f2f;
}

.emotion-3 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 4px;
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-3:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-3:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-3.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-3.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-3.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 1px;
  padding: 8.5px 14px;
}

.emotion-4::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4:focus {
  outline: 0;
}

.emotion-4:invalid {
  box-shadow: none;
}

.emotion-4::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-4.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-4:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-4:-webkit-autofill {
  border-radius: inherit;
}

.emotion-5 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-6 {
  float: unset;
  overflow: hidden;
  display: block;
  width: auto;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 100%;
  -webkit-transition: max-width 100ms cubic-bezier(0.0, 0, 0.2, 1) 50ms;
  transition: max-width 100ms cubic-bezier(0.0, 0, 0.2, 1) 50ms;
  white-space: nowrap;
}

.emotion-6>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  opacity: 0;
  visibility: visible;
}

<div>
  <form
    novalidate=""
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth emotion-0"
    >
      <div
        class="MuiFormControl-root MuiTextField-root  emotion-1"
        errors="[object Object]"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled emotion-2"
          data-shrink="true"
          for="mui-4"
          id="mui-4-label"
        >
          Name
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall emotion-3"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall emotion-4"
            data-testid="TextFieldHF__input"
            id="mui-4"
            name="name"
            type="text"
            value="test data"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline emotion-5"
          >
            <legend
              class="emotion-6"
            >
              <span>
                Name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <input
      data-testid="TextFieldHF__submit"
      type="submit"
    />
  </form>
</div>
`;

exports[`<TextFieldHF /> required error should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-1 .TextFieldBase-readOnly .MuiInputBase-root {
  background-color: #fafafa;
}

.emotion-2 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(100% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, 9px) scale(1);
  -moz-transform: translate(14px, 9px) scale(1);
  -ms-transform: translate(14px, 9px) scale(1);
  transform: translate(14px, 9px) scale(1);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: none;
}

.emotion-2.Mui-focused {
  color: #1976d2;
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-2.Mui-error {
  color: #d32f2f;
}

.emotion-3 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 4px;
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-3:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-3:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-3.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-3.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-3.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 1px;
  padding: 8.5px 14px;
}

.emotion-4::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4:focus {
  outline: 0;
}

.emotion-4:invalid {
  box-shadow: none;
}

.emotion-4::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-4.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-4:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-4:-webkit-autofill {
  border-radius: inherit;
}

.emotion-5 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-6 {
  float: unset;
  overflow: hidden;
  display: block;
  width: auto;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 0.01px;
  -webkit-transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  white-space: nowrap;
}

.emotion-6>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  opacity: 0;
  visibility: visible;
}

<div>
  <form
    novalidate=""
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth emotion-0"
    >
      <div
        class="MuiFormControl-root MuiTextField-root  emotion-1"
        errors="[object Object]"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary emotion-2"
          data-shrink="false"
          for="mui-2"
          id="mui-2-label"
        >
          Name
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall emotion-3"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall emotion-4"
            data-testid="TextFieldHF__input"
            id="mui-2"
            name="name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline emotion-5"
          >
            <legend
              class="emotion-6"
            >
              <span>
                Name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <input
      data-testid="TextFieldHF__submit"
      type="submit"
    />
  </form>
</div>
`;
